INTPYTHON-580 Design and Implement MongoDBVectorSearchTool

blink1073 opened the pull request:
Integration tests: https://github.com/mongodb-labs/ai-ml-pipeline-testing/pull/71

blink1073 committed:
INTPYTHON-580 Design and Implement MongoDBVectorSearchTool

blink1073 committed:
add implementation

blink1073 committed:
wip

blink1073 committed:
wip

blink1073 committed:
finish tests

blink1073 committed:
add todo

blink1073 committed:
refactor to wrap langchain-mongodb

blink1073 committed:
cleanup

caseyclements commented on crewai_tools/tools/mongodb_vector_search_tool/README.md:
```diff
@@ -0,0 +1,85 @@
+    collection_name='example_collections',
+    connection_string="<your_mongodb_connection_string>",
+    query_config=query_config,
+    index_name="my_vector_index",
```
I'd call this vector_index_name. It is explicit and will avoid backwards compatibility issues is this gets adoption and we want access to other search types.

blink1073 replied:
Done

caseyclements commented on crewai_tools/tools/mongodb_vector_search_tool/README.md:
```diff
@@ -0,0 +1,85 @@
+from crewai_tools import MongoDBVectorSearchConfig, MongoDBVectorSearchTool
+
+# Setup custom embedding model and customize the parameters.
+query_config = MongoDBVectorSearchConfig(limit=10, oversampling_factor=2)
```
It feels like you could combine the two examples. "MongoDBVectorSearchTool provides a number of configurable parameters. The kwarg query_config takes a MongoDBVectorSearchConfig. For example....

On the vector index, is this automatically created? Is it clear what will you vectorized? It's worth noting that embedding models can embed any text, from plain text to embedded json.

blink1073 replied:
I like showing the simplest case so they can copy-paste and get rolling.  No, the vector index has to be explicitly created using `create_vector_search_index`.  I don't follow the embedded json part, the type annotation for [add_texts](https://github.com/langchain-ai/langchain-mongodb/blob/3214d2f02753fb58aee70c97cf486e4ef2a97bb9/libs/langchain-mongodb/langchain_mongodb/vectorstores.py#L299) is `texts: Iterable[str]`.

caseyclements replied:
Just thoughts. This looks great. Wrapping langchain_mongodb was a brilliant move.

caseyclements commented on crewai_tools/tools/mongodb_vector_search_tool/vector_search.py:
```diff
@@ -0,0 +1,217 @@
+
+    query: str = Field(
+        ...,
+        description="The query to search retrieve relevant information from the MongoDB database. Pass only the query, not the question.",
```
What's the difference between query and question? Is that a CrewAI thing?

blink1073 replied:
I was following prior examples.

caseyclements reviewed (approved):
LGTM with comments.

blink1073 committed:
address review

blink1073 committed:
Merge branch 'main' of github.com:crewAIInc/crewAI-tools into INTPYTHON-580

blink1073 committed:
Fix usage of EnvVar class

lucasgomide committed:
build: upgrade mcpadapt version (#323)

This version contains a dedicated fix fro CrewAIAdapter when schema doesn't allow null

lorenzejay committed:
chore: update version to 0.47.0 in pyproject.toml and uv.lock

tonykipkemboi committed:
Merge pull request #324 from crewAIInc/lorenze/version-0.47.0

blink1073 committed:
inline code

blink1073 committed:
merge from master

blink1073 committed:
lint

blink1073 committed:
lint

blink1073 committed:
fix usage of SearchIndexModel

lorenzejay committed:
Refactor: Update EnvVar import path and remove unused tests.utils module

- Changed import of EnvVar from tests.utils to crewai.tools in multiple files.
- Updated README.md for MongoDB vector search tool with additional context.
- Modified subprocess command in vector_search.py for package installation.
- Cleaned up test_generate_tool_specs.py to improve mock patching syntax.
- Deleted unused tests/utils.py file.

lucasgomide committed:
fix: ensure the entire file will be read when the start_line is None (#325)

blink1073 committed:
update the crewai dep and the lockfile

lorenzejay committed:
chore: update version to 0.47.1 and upgrade crewai dependency to 0.126.0 (#327)

lorenzejay committed:
refactor: remove token validation from EnterpriseActionKitToolAdapter… (#331)

* refactor: remove token validation from EnterpriseActionKitToolAdapter and CrewaiEnterpriseTools

This commit simplifies the initialization of the EnterpriseActionKitToolAdapter and CrewaiEnterpriseTools by removing the explicit validation for the enterprise action token. The token can now be set to None without raising an error, allowing for more flexible usage.

* added loggers for monitoring

* fixed typo

lucasgomide committed:
Support to collect extra package dependencies of Tools (#330)

* feat: add explictly package_dependencies in the Tools

* feat: collect package_dependencies from Tool to add in tool.specs.json

* feat: add default value in run_params Tool' specs

* fix: support get boolean values

This commit also refactor test to make easier define newest attributes into a Tool

lucasgomide committed:
refactor: renaming init_params and run_params to reflect their schema. (#332) (#333)

We’re currently using the JSON Schema standard for these fields

lucasgomide committed:
feat: mapping explicitly tool environment variables (#338)

lucasgomide committed:
feat: Add ToolCollection class for named tool access (#339)

This change allows accessing tools by name (tools["tool_name"]) in addition to
index (tools[0]), making it more intuitive and convenient to work with multiple
tools without needing to remember their position in the list

lucasgomide committed:
fix: add support for case-insensitive Enterprise filter (#340)

oxy-rostyslav committed:
Add Oxylabs Web Scraping tools (#312)

* Add Oxylabs tools

* Review updates

* Add package_dependencies attribute

lucasgomide committed:
feat: support api_key fallback to EXA_API_KEY env-var (#341)

lucasgomide committed:
Support to filter available MCP Tools (#345)

* feat: support to complex filter on ToolCollection

* refactor: use proper tool collection methot to filter tool in CrewAiEnterpriseTools

* feat: allow to filter available MCP tools

lorenzejay committed:
Lorenze/better env vars setup enterprise tools (#343)

* refactor: remove token validation from EnterpriseActionKitToolAdapter and CrewaiEnterpriseTools

This commit simplifies the initialization of the EnterpriseActionKitToolAdapter and CrewaiEnterpriseTools by removing the explicit validation for the enterprise action token. The token can now be set to None without raising an error, allowing for more flexible usage.

* added loggers for monitoring

* fixed typo

* fix: enhance token handling in EnterpriseActionKitToolAdapter and CrewaiEnterpriseTools

This commit improves the handling of the enterprise action token by allowing it to be fetched from environment variables if not provided. It adds checks to ensure the token is set before making API requests, enhancing robustness and flexibility.

* removed redundancy

* test: add new test for environment token fallback in CrewaiEnterpriseTools

This update introduces a new test case to verify that the environment token is used when no token is provided during the initialization of CrewaiEnterpriseTools. Additionally, minor formatting adjustments were made to existing assertions for consistency.

* test: update environment token test to clear environment variables

This change modifies the test for CrewaiEnterpriseTools to ensure that the environment variables are cleared before setting the test token. This ensures a clean test environment and prevents potential interference from other tests.

* drop redundancy

lorenzejay committed:
feat: add support for parsing actions list from environment variables (#346)

* feat: add support for parsing actions list from environment variables

This commit introduces a new function, _parse_actions_list, to handle the parsing of a string representation of a list of tool names from environment variables. The CrewaiEnterpriseTools now utilizes this function to filter tools based on the parsed actions list, enhancing flexibility in tool selection. Additionally, a new test case is added to verify the correct usage of the environment actions list.

* test: simplify environment actions list test setup

This commit refactors the test for CrewaiEnterpriseTools to streamline the setup of environment variables. The environment token and actions list are now set in a single patch.dict call, improving readability and reducing redundancy in the test code.

lorenzejay committed:
new version of tools (#347)

lucasgomide committed:
build: fix command to generate tools specs on CI (#350)

lorenzejay committed:
refactor: update Firecrawl tools to improve configuration and error handling (#351)

- Added TYPE_CHECKING imports for FirecrawlApp to enhance type safety.
- Updated configuration keys in FirecrawlCrawlWebsiteTool and FirecrawlScrapeWebsiteTool to camelCase for consistency.
- Introduced error handling in the _run methods of both tools to ensure FirecrawlApp is properly initialized before usage.
- Adjusted parameters passed to crawl_url and scrape_url methods to use 'params' instead of unpacking the config dictionary directly.

thiagomoretto committed:
Mapping required env vars of more tools (#353)

lucasgomide committed:
update tool.spec.json (#354)

emmanuel-ferdman committed:
fix: update Pydantic schema access (#337)

Signed-off-by: Emmanuel Ferdman <emmanuelferdman@gmail.com>

lorenzejay committed:
refactor: enhance schema handling in EnterpriseActionTool (#355)

* refactor: enhance schema handling in EnterpriseActionTool

- Extracted schema property and required field extraction into separate methods for better readability and maintainability.
- Introduced methods to analyze field types and create Pydantic field definitions based on nullability and requirement status.
- Updated the _run method to handle required nullable fields, ensuring they are set to None if not provided in kwargs.

* refactor: streamline nullable field handling in EnterpriseActionTool

- Removed commented-out code related to handling required nullable fields for clarity.
- Simplified the logic in the _run method to focus on processing parameters without unnecessary comments.

lorenzejay committed:
chore: update crewai dependency version to 0.134.0 in pyproject.toml and uv.lock (#356)

lorenzejay committed:
chore: bump version to 0.49.0 in pyproject.toml and uv.lock (#357)

lorenzejay committed:
Merge branch 'main' of github.com:crewAIInc/crewAI-tools into INTPYTHON-580

lorenzejay committed:
chore: update package versions and dependencies in uv.lock

- Removed `auth0-python` package.
- Updated `crewai` version to 0.140.0 and adjusted its dependencies.
- Changed `json-repair` version to 0.25.2.
- Updated `litellm` version to 1.72.6.
- Modified dependency markers for several packages to improve compatibility with Python versions.

capemox committed:
Add Couchbase as a tool (#264)

* - Added CouchbaseFTSVectorStore as a CrewAI tool.
- Wrote a README to setup the tool.
- Wrote test cases.
- Added Couchbase as an optional dependency in the project.

* Fixed naming in some places. Added docstrings. Added instructions on how to create a vector search index.

* Fixed pyproject.toml

* error handling and response format

- Removed unnecessary ImportError for missing 'couchbase' package.
- Changed response format from a concatenated string to a JSON array for search results.
- Updated error handling to return error messages instead of raising exceptions in certain cases.
- Adjusted tests to reflect changes in response format and error handling.

* Update dependencies in pyproject.toml and uv.lock

- Changed pydantic version from 2.6.1 to 2.10.6 in both pyproject.toml and uv.lock.
- Updated crewai-tools version from 0.42.2 to 0.42.3 in uv.lock.
- Adjusted pydantic-core version from 2.33.1 to 2.27.2 in uv.lock, reflecting the new pydantic version.

* Removed restrictive pydantic version and updated uv.lock

* synced lockfile

* regenerated lockfile

* updated lockfile

* regenerated lockfile

* Update tool specifications for

* Fix test cases

---------

Co-authored-by: AayushTyagi1 <tyagiaayush5@gmail.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

lorenzejay committed:
refactor: improve MongoDB vector search tool with enhanced error handling and new dimensions field

- Added logging for error handling in the _run method and during client cleanup.
- Introduced a new 'dimensions' field in the MongoDBVectorSearchConfig for embedding vector size.
- Refactored the _run method to return JSON formatted results and handle exceptions gracefully.
- Cleaned up import statements and improved code readability.

blink1073 committed:
address review

blink1073 committed:
update tests

blink1073 committed:
Merge branch 'main' of github.com:crewAIInc/crewAI-tools into INTPYTHON-580

blink1073 committed:
debug

blink1073 committed:
fix test

blink1073 committed:
fix test

blink1073 committed:
fix test

blink1073 committed:
support azure openai
